packages/astro/src/runtime/server/index.ts

```diff
@@ -545,7 +545,11 @@ Make sure to use the static attribute syntax (\`${key}={value}\`) instead of the
 
 	// support "class" from an expression passed into an element (#782)
 	if (key === 'class:list') {
-		return markHTMLString(` ${key.slice(0, -5)}="${toAttributeString(serializeListValue(value))}"`);
+		const listValue = toAttributeString(serializeListValue(value));
```

Comment:
is there a reason that we don't use clhttps://www.npmjs.com/package/clsx here? It looks like we implemented this ourselves, but clsx is so battle-tested (and zero dep) I can't think of a good reason not to use it here

Not to tackle here, just calling this out to reduce our bug risk and maintenance costs

Reply:
I don't know the original reason for not using `clsx` directly. I'll keep an eye out for any more issues that come up... certainly happy to switch in the future. For now would love to get this hotfix out.